src/Simulation/CsharpGeneration/RewriteStep.fs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.CsharpGeneration
+
+open System
+open System.Collections.Generic
+open System.IO
+open Microsoft.Quantum.QsCompiler
+open Microsoft.Quantum.QsCompiler.CsharpGeneration
+open Microsoft.Quantum.QsCompiler.ReservedKeywords
+open Microsoft.Quantum.QsCompiler.SyntaxTree
+open Microsoft.Quantum.QsCompiler.Transformations.BasicTransformations
+
+
+type Emitter() =
+
+    let _AssemblyConstants = new Dictionary<string, string>()
+
+    static member internal SupportUnitTests = "CsharpGeneration/TestProject"
+
+    interface IRewriteStep with
+
+        member this.Name = "CsharpGeneration"
+        member this.Priority = -1 // doesn't matter because this rewrite step is the only one in the dll
+        member this.AssemblyConstants = _AssemblyConstants :> IDictionary<string, string> 
+        
+        member this.ImplementsPreconditionVerification = false
+        member this.ImplementsPostconditionVerification = false
+        member this.ImplementsTransformation = true
+
+        member this.PreconditionVerification _ = NotImplementedException() |> raise
+        member this.PostconditionVerification _ = NotImplementedException() |> raise
+        
+        member this.Transformation (compilation, transformed) = 
+            let step = this :> IRewriteStep
+            let dir = step.AssemblyConstants.TryGetValue AssemblyConstants.OutputPath |> function
+                | true, outputFolder when outputFolder <> null -> outputFolder
+                | _ -> step.Name
+            let allSources = 
+                GetSourceFiles.Apply compilation.Namespaces 
+                |> Seq.filter (fun fileName -> (fileName.Value |> Path.GetFileName).StartsWith "Microsoft.Quantum" |> not)
+            for source in allSources do
+                let content = compilation.Namespaces |> SimulationCode.generate source
+                CompilationLoader.GeneratedFile(source, dir, ".g.cs", content) |> ignore
```

Comment:
Remind me, why is this necessary? Shouldn't all that is needed is to write the file to disk?

Reply:
Waht do you mean? That is what it's doing?

Reply:
yeah, i was asking about `GeneratedFile`. It was unclear why that logic to write the file to disk was wrapped in that method.